### ISSUE TITLE
raise condiation in j.code.execute

### ISSUE THREAD

**Issue by grimpy** [Jumpscale/jumpscale_core7#143]
````python
traceback (most recent call last): 
~ File "/usr/local/bin/jscode", line 103, in <module>  _, url, _ = j.do.execute("cd %s;git config --get remote.origin.url" % path) 
~ File "/opt/jumpscale7/lib/JumpScale/InstallTools.py", line 1080, in execute  (output2,error2) = p.communicate() 
~ File "/opt/jumpscale7/lib/subprocess.py", line 799, in communicate  return self._communicate(input) 
~ File "/opt/jumpscale7/lib/subprocess.py", line 1401, in _communicate  stdout, stderr = self._communicate_with_poll(input) 
~ File "/opt/jumpscale7/lib/subprocess.py", line 1481, in _communicate_with_poll  close_unregister_and_remove(fd) 
~ File "/opt/jumpscale7/lib/subprocess.py", line 1438, in close_unregister_and_remove  fd2file[fd].close() 
~ IOError: close() called during concurrent operation on the same file object.
```

**Comment by grimpy** [comment]
This happens atleast on the ays_unstable branch

Write a test to reproduce this.
Then see how to fix.

**Comment by Hamdy** [comment]
fixed .. thanks to @muhamadazmy

**Comment by despiegk** [comment]
cool

On Mon, Nov 9, 2015 at 2:17 PM, Hamdy <notifications@github.com> wrote:

> Closed #143 <https://github.com/Jumpscale/jumpscale_core7/issues/143>.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Jumpscale/jumpscale_core7/issues/143#event-458813490>.
>



-- 
------------------------
Kristof De Spiegeleer
+971525609014
+201206927877
+32 475405474
skype: despiegk
